examples/llm_plus_gnn/rag_generate.py

```diff
@@ -0,0 +1,96 @@
+# %%
+from itertools import chain
+
+import pandas as pd
+import torch
+import tqdm
+from profiling_utils import create_remote_backend_from_triplets
+from rag_feature_store import SentenceTransformerFeatureStore
+from rag_graph_store import NeighborSamplingRAGGraphStore
+
+from torch_geometric.data import Data, get_features_for_triplets_groups
+from torch_geometric.datasets import UpdatedWebQSPDataset
+from torch_geometric.datasets.updated_web_qsp_dataset import preprocess_triplet
+from torch_geometric.loader import RAGQueryLoader
+from torch_geometric.nn.nlp import SentenceTransformer
+
+# %%
+ds = UpdatedWebQSPDataset("small_dataset", force_reload=True, limit=20)
+
+# %%
+triplets = chain.from_iterable(d['graph'] for d in ds.raw_dataset)
+
+# %%
+questions = ds.raw_dataset['question']
+questions
+
+# %%
+ground_truth_graphs = get_features_for_triplets_groups(
+    ds.indexer, (d['graph'] for d in ds.raw_dataset),
+    pre_transform=preprocess_triplet)
+num_edges = len(ds.indexer._edges)
+
+# %%
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+model = SentenceTransformer(
+    model_name='sentence-transformers/all-roberta-large-v1').to(device)
+
+# %%
+fs, gs = create_remote_backend_from_triplets(
+    triplets=triplets, node_embedding_model=model,
+    node_method_to_call="encode", path="backend",
+    pre_transform=preprocess_triplet, node_method_kwargs={
+        "batch_size": 256
+    }, graph_db=NeighborSamplingRAGGraphStore,
+    feature_db=SentenceTransformerFeatureStore).load()
+
+# %%
+query_loader = RAGQueryLoader(data=(fs, gs), seed_nodes_kwargs={"k_nodes": 10},
+                              seed_edges_kwargs={"k_edges": 10},
+                              sampler_kwargs={"num_neighbors": [40] * 3})
+
```

Comment:
lets put this under examples/llm_plus_gnn/data_handling/
(open to suggestions for better name, but something like that)

Reply:
This is part of the PoC for the FeatureStore and GraphStore defined here for now, but if people think this should be applied generally, I'm open to making the changes to enable that.